demos/cairo/cairo_clock/clock.d

```diff
@@ -35,12 +36,14 @@ public:
 	this()
 	{
 		//Attach our expose callback, which will draw the window.
-		addOnDraw(&drawCallback);
+		setDrawFunc ((area, cairo, w, h, data) {
+            (cast(Clock)data).drawCallback (new Context (cairo));
+        }, cast(void *)this, null);
 	}
 
 protected:
 	//Override default signal handler:
-	bool drawCallback(Scoped!Context cr, Widget widget)
+	bool drawCallback(Context cr)
```

Comment:
Without `Scoped` you will either need to free things manually every drawCallback, or you will have a large memory leak, this because the D GC doesn't see the large `Context` associated with the D object and so it will take a long time for the contexts to be collected.